src/components/Articles/ViewArticle.js

```diff
@@ -1,41 +1,156 @@
 import React, { Component } from 'react';
-import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
 import { bindActionCreators } from 'redux';
-import { fetchSpecificArticle } from '../../actions/articleActions';
+import PropTypes from 'prop-types';
+import {
+  fetchSpecificArticle,
+  updateArticle,
+} from '../../actions/articleActions';
 import ReadArticle from './ReadArticle';
+import NewArticleForm from './NewArticleForm';
 
 export class ViewArticle extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: '',
+      description: '',
+      body: '',
+      isEditing: false,
+    };
+  }
+
   componentDidMount() {
-    const { match: { params: { slug } }, fetchSpecificArticle } = this.props;
+    const { fetchSpecificArticle } = this.props;
+    const {
+      match: {
+        params: { slug },
+      },
+    } = this.props;
     fetchSpecificArticle(slug);
   }
 
-  render() {
-    const { articlePayload, articlePayload: { article } } = this.props;
+  componentWillReceiveProps(nextProps) {
+    const { articlePayload } = nextProps;
+    const { article: { title, body, description } } = articlePayload;
+    this.setState({ title });
+    this.setState({ body });
+    this.setState({ description });
```

Comment:
Use one `setState` statement.

Reply:
This piece of code has been refactored as advised